### PR TITLE
Launch Nexus using a self-signed x.509 certificate

### PR DESCRIPTION
**Pull request by smklein**
Part of https://github.com/oxidecomputer/omicron/issues/249

This PR forces Nexus's external interface to be served via HTTPS when deployed by the sled-agent.

- The packaging system expects to find these certificates within `./out/certs`, named `cert.pem` and `key.pem`.
- `./tools/create_self_signed_cert.sh` is capable of creating a self-signed certificate.

### CODE REVIEW

**Review by davepacheco — COMMENTED** [review]
This looks good.  Two things:

- If somebody forgets to create the self-signed certificate but still runs builds the image, what's the failure mode?  It'd be nice if that fails building the image rather than failing at runtime.
- I wonder if we should update the example config files and the simulated run instructions so that people are doing this in that workflow too.  But that doesn't have to be in this PR!

**Comment by smklein** [comment]
> This looks good. Two things:
> 
> * If somebody forgets to create the self-signed certificate but still runs builds the image, what's the failure mode?  It'd be nice if that fails building the image rather than failing at runtime.

The packaging tools complain when constructing the Nexus zone image - so it is not a runtime failure. I wanted to make the certificates mandatory for running in "real" nexus to make it very difficult to "not have them at runtime, somehow".

> * I wonder if we should update the example config files and the simulated run instructions so that people are doing this in that workflow too.  But that doesn't have to be in this PR!

Sounds good

File: tools/create_self_signed_cert.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -eu
+
+# Creates a self-signed certificate.
+#
+# For those with access, certificates are available in:
+#
+# https://github.com/oxidecomputer/configs/tree/master/nginx/ssl/wildcard.oxide-preview.com
```
Comment (augustuswm) [inline tools/create_self_signed_cert.sh:9]: This location is will change eventually as a some things get decoupled. Not to block this PR, but for future reference.

**Review by augustuswm — APPROVED** [review]

File: tools/create_self_signed_cert.sh (hunk 1 above)
Comment (smklein) [reply tools/create_self_signed_cert.sh:9]: Thanks for the heads up. The flow for importing a non-self-signed cert is clearly half-baked; I needed to pull them manually.

Until we have a good authentication story for it, the manual approach seems better than something insecure, but I'd like to remedy that when we can.

**Pull request merged**